lib/response-types/token-response-type.ts

```diff
@@ -0,0 +1,97 @@
+import { InvalidArgumentError } from '../errors';
+import { ImplicitGrantType } from '../grant-types';
+import { Client, Model, User } from '../interfaces';
+import { Request } from '../request';
+
+export class TokenResponseType {
+  accessToken: string;
+  accessTokenLifetime: number;
+  model: Model;
+  constructor(options: any = {}) {
+    if (!options.accessTokenLifetime) {
```

Comment:
Type here should probably be `{model: Model, accessTokenLifetime: number}`, and the default removed.

This also doesn't check that `options.model` is provided here, while it does in other places.